tools/distpartitioning/data_shuffle.py

```diff
@@ -92,12 +92,19 @@ def gen_node_data(rank, world_size, node_part_ids, ntid_ntype_map, nid_schema_ma
     gnid_start = 0
     gnid_end = 0
     for ntypeid in range(len(ntid_ntype_map)):
-        ntype_name = ntid_ntype_map[str(ntypeid)]
-        ntype_info = nid_schema_map[ntype_name]
-        type_start = int(ntype_info["data"][0][1])
-        type_end = int(ntype_info["data"][-1][2])
+        ntype_name = ntid_ntype_map[ntypeid]
+        ntype_info = schema_map[constants.STR_NUM_NODES_PER_CHUNK]
 
-        gnid_end += type_end
+        tid_info = ntype_info[ntypeid] #array of no.... chunks per partition
+        
+        tid_start = np.cumsum([0] + tid_info[:-1])
+        tid_end = np.cumsum(tid_info)
+        tid_ranges = list(zip(tid_start, tid_end))
+
+        type_start = tid_ranges[0][0]
+        type_end = tid_ranges[-1][1]
```

Comment:
i feel you should define a function for converting the number of nodes to an ID range.

Reply:
Agreed. This logic is used in multiple places (convert_partition.py, data_shuffle.py and dataset_utils.py). Now added a new function in the utils.py to compte the type-ids and global-ids of nodes and edges and this new function is used everywhere such functionality is needed.